pkg/verifier/aws/aws_verifier.go

```diff
@@ -325,3 +325,129 @@ func (a *AwsVerifier) writeDebugLogs(log string) {
 	a.Output.AddDebugLogs(log)
 	a.Logger.Debug(context.TODO(), log)
 }
+
+// CreateSecurityGroup creates a security group with the specified name and cluster tag key in a specified VPC
+func (a *AwsVerifier) CreateSecurityGroup(ctx context.Context, tags map[string]string, name, vpcId string) (*ec2.CreateSecurityGroupOutput, error) {
+	input := &ec2.CreateSecurityGroupInput{
+		GroupName:   awsTools.String(name),
+		VpcId:       &vpcId,
+		Description: awsTools.String("osd-network-verifier security group"),
+	}
+	a.writeDebugLogs("Creating a Security group")
+	output, err := a.AwsClient.CreateSecurityGroup(ctx, input)
+	if err != nil {
+		return &ec2.CreateSecurityGroupOutput{}, err
+	}
+
+	a.writeDebugLogs(fmt.Sprintf("Waiting for the Security Group to exist: %s", *output.GroupId))
+	// Wait up to 1 minutes for the security group to exist
+	waiter := ec2.NewSecurityGroupExistsWaiter(a.AwsClient)
+	if err := waiter.Wait(ctx, &ec2.DescribeSecurityGroupsInput{GroupIds: []string{*output.GroupId}}, 1*time.Minute); err != nil {
+		a.writeDebugLogs(fmt.Sprintf("Error waiting for the security group to exist: %s, attempting to delete the Security Group", *output.GroupId))
+		if _, err := a.AwsClient.DeleteSecurityGroup(ctx, &ec2.DeleteSecurityGroupInput{
+			GroupId: output.GroupId,
+		}); err != nil {
+			return &ec2.CreateSecurityGroupOutput{}, handledErrors.NewGenericError(err)
+		}
+		return &ec2.CreateSecurityGroupOutput{}, fmt.Errorf("deleted %s after timing out waiting for security group to exist", *output.GroupId)
+	}
+
+	a.Logger.Info(ctx, "Created security group with ID: %s", *output.GroupId)
+	if err := a.createTags(tags, *output.GroupId); err != nil {
+		// Unable to tag the instance
+		if _, err := a.AwsClient.DeleteSecurityGroup(ctx, &ec2.DeleteSecurityGroupInput{
```

Comment:
I think it would read cleaner here with the if before the err statement instead of the before the return parameters same as the others